### PR TITLE
feat: OR query support

### PR DESCRIPTION
**Pull request by kolea2**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java [hunk 1]
```diff
@@ -223,6 +224,89 @@ private <T> List<T> makeResultsCopy(QueryResults<T> scResults) {
...
+      QueryResults<Entity> results3 = DATASTORE.run(orQueryNested);
+      assertTrue(results3.hasNext());
+      assertEquals(ENTITY2, results3.next());
+      assertFalse(results3.hasNext());
+
+      // age == 20 && (name == Bob || name == Dan)
+      CompositeFilter nestedOr2 =
+          CompositeFilter.or(PropertyFilter.eq("name", "Dan"), PropertyFilter.eq("name", "Bob"));
+      CompositeFilter andFilter = CompositeFilter.and(PropertyFilter.eq("age", 20), nestedOr2);
+      CompositeFilter ancestorAndFilter =
```
Comment (danieljbruce) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:292]: Just a reminder that in the `nodejs-datastore` PRs there has been no change to `hasAncestor` functionality and deprecating it is in a separate PR so this should (and does) line up with the current `nodejs-datastore` PR.

**Review by danieljbruce — APPROVED** [review]
LGTM

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java (hunk 1 above)
Comment (kolea2) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:292]: yes - this PR does not include any changes here, this is the current way to do this kind of filter.

**Review by jainsahab — CHANGES_REQUESTED** [review]

**Comment by jainsahab** [comment]
just one minor comment, apart from that it's LGTM

**Review by jainsahab — APPROVED** [review]

**Pull request merged**